base/shared/src/main/scala/scalaz/data/eval.scala

```diff
@@ -0,0 +1,127 @@
+package scalaz
+package data
+
+import scalaz.Predef._
+import scalaz.prop.IsCovariant
+import tc._
+
+sealed abstract class EvalModule {
+  type Eval[A]
+
+  /** Call by value */
+  def value[A](a: A): Eval[A]
+
+  /** Call by need */
+  def need[A](a: () => Eval[A]): Eval[A]
+
+  /** Call by name */
+  def name[A](a: () => Eval[A]): Eval[A]
+
+  def eval[A](thunk: Eval[A]): A
+
+  private[data] def isCovariant: IsCovariant[Eval]
+}
+
+object EvalModule {
+  implicit def isCovariant: IsCovariant[Eval] = Eval.isCovariant
+
+  implicit def name[A]: Name[Eval[A]] = instanceOf[NameClass[Eval[A]]](Eval.name(_))
+
+  implicit def need[A]: Need[Eval[A]] = instanceOf[NeedClass[Eval[A]]](Eval.need(_))
+}
+
+object Eval {
+  val module: EvalModule = EvalModuleImpl
+
+  def unapply[A](thunk: Eval[A]): Evaluated[A] = new Evaluated(Eval.eval(thunk))
+
+  import scala.language.implicitConversions
+  @scala.inline
+  implicit def evalModule(eval: Eval.type): module.type = eval.module
```

Comment:
so that Eval can expose both unapply AND EvalModule api: is that a valid use case?